File: examples/apple-demo/Sources/UICheckAppleDemoSupport/DemoView.swift
import AppKit

public struct UICheckAppleDemoView {
  public let window: NSWindow
  public let content: UICheckAppleDemoContent
}

public struct UICheckAppleDemoContent {
  public let screen: NSView
  public let title: NSTextField
  public let summary: NSView
  public let items: NSView
  public let status: NSView
  public let details: NSView
  public let submit: NSButton
}

@MainActor
public func createUICheckAppleDemoView() -> UICheckAppleDemoView {
  let window = NSWindow(contentRect: NSRect(x: 0, y: 0, width: 390, height: 844), styleMask: [.titled, .closable], backing: .buffered, defer: false)
  window.title = "UICheck Apple Demo"
  let content = createUICheckAppleDemoContent()
  window.contentView = content.screen
  window.makeKeyAndOrderFront(nil)
  return UICheckAppleDemoView(window: window, content: content)
}

@MainActor
public func createUICheckAppleDemoContent() -> UICheckAppleDemoContent {
  let screen = NSView(frame: NSRect(x: 0, y: 0, width: 390, height: 844))
  screen.identifier = NSUserInterfaceItemIdentifier("screen")
  screen.wantsLayer = true
  screen.layer?.backgroundColor = NSColor(calibratedRed: 0.97, green: 0.98, blue: 0.99, alpha: 1).cgColor

  let header = NSView(frame: NSRect(x: 0, y: 766, width: 390, height: 78))
  header.wantsLayer = true
  header.layer?.backgroundColor = NSColor(calibratedRed: 0.067, green: 0.094, blue: 0.153, alpha: 1).cgColor
  let eyebrow = label("UICheck Apple", frame: NSRect(x: 24, y: 44, width: 180, height: 18), size: 13, color: NSColor(calibratedRed: 0.576, green: 0.773, blue: 0.992, alpha: 1), bold: true)
  let title = label("Checkout screen", frame: NSRect(x: 24, y: 12, width: 248, height: 28), size: 22, color: .white, bold: true)
  let badge = label("apple", frame: NSRect(x: 310, y: 23, width: 56, height: 24), size: 12, color: .white, bold: true)
  badge.alignment = .center
  badge.wantsLayer = true
  badge.layer?.borderColor = NSColor(calibratedWhite: 1, alpha: 0.36).cgColor
  badge.layer?.borderWidth = 1
  badge.layer?.cornerRadius = 12
  header.addSubview(eyebrow)
  header.addSubview(title)
  header.addSubview(badge)

  let summary = card(frame: NSRect(x: 14, y: 686, width: 362, height: 66), title: "Registered ref summary", text: "MCP reads runtime boxes, text, testID and labels.")
  let items = orderCard(frame: NSRect(x: 14, y: 586, width: 362, height: 92))
  let status = card(frame: NSRect(x: 14, y: 512, width: 362, height: 66), title: "Ready for MCP inspection", text: "This real demo has 100+ inspectable nodes.")
  let details = detailsPanel(frame: NSRect(x: 14, y: 134, width: 362, height: 370))
  let hint = label("MCP can inspect all elements or a selected target.", frame: NSRect(x: 24, y: 90, width: 342, height: 18), size: 11, color: NSColor(calibratedRed: 0.059, green: 0.463, blue: 0.431, alpha: 1), bold: true)
  let hintBox = NSView(frame: NSRect(x: 14, y: 82, width: 362, height: 34))
  hintBox.wantsLayer = true
  hintBox.layer?.backgroundColor = NSColor(calibratedRed: 0.925, green: 0.996, blue: 1, alpha: 1).cgColor
  hintBox.layer?.cornerRadius = 12
  hintBox.addSubview(hint)
  let submit = NSButton(title: "Submit order", target: nil, action: nil)
  submit.identifier = NSUserInterfaceItemIdentifier("submit-button")
  submit.frame = NSRect(x: 14, y: 28, width: 362, height: 40)
  submit.isBordered = false
  submit.wantsLayer = true
  submit.layer?.backgroundColor = NSColor(calibratedRed: 0.659, green: 0.333, blue: 0.969, alpha: 1).cgColor
  submit.layer?.cornerRadius = 10
  submit.attributedTitle = NSAttributedString(
    string: "Submit order",
    attributes: [
      .foregroundColor: NSColor.white,
      .font: NSFont.boldSystemFont(ofSize: 14)
    ]
  )

  [header, summary, items, status, details, hintBox, submit].forEach(screen.addSubview)
  return UICheckAppleDemoContent(screen: screen, title: title, summary: summary, items: items, status: status, details: details, submit: submit)
}

@MainActor
private func card(frame: NSRect, title: String, text: String) -> NSView {
  let view = shell(frame)
  view.addSubview(label(title, frame: NSRect(x: 10, y: 38, width: frame.width - 20, height: 18), size: 13, color: dark, bold: true))
  view.addSubview(label(text, frame: NSRect(x: 10, y: 14, width: frame.width - 20, height: 18), size: 11, color: muted, bold: false))
  return view
}

@MainActor
private func orderCard(frame: NSRect) -> NSView {
  let view = shell(frame)
  view.addSubview(label("Order items", frame: NSRect(x: 10, y: 64, width: frame.width - 20, height: 18), size: 13, color: dark, bold: true))
  view.addSubview(orderRow("Starter license", "$19", y: 44, width: frame.width))
  view.addSubview(orderRow("Team add-on", "$8", y: 28, width: frame.width))
  view.addSubview(orderRow("Total", "$27", y: 10, width: frame.width, bold: true))
  return view
}

@MainActor
private func detailsPanel(frame: NSRect) -> NSView {
  let view = shell(frame, padding: 8)
  view.addSubview(label("Runtime detail matrix", frame: NSRect(x: 8, y: 344, width: frame.width - 16, height: 18), size: 13, color: dark, bold: true))
  for line in 0..<17 {
    for column in 0..<2 {
      let number = line * 2 + column + 1
      let x = 8 + CGFloat(column) * 174
      let y = 322 - CGFloat(line) * 19
      view.addSubview(detailCell(number, frame: NSRect(x: x, y: y, width: 166, height: 16)))
    }
  }
  return view
}

@MainActor
private func detailCell(_ number: Int, frame: NSRect) -> NSView {
  let cell = NSView(frame: frame)
  cell.wantsLayer = true
  cell.layer?.backgroundColor = NSColor(calibratedRed: 0.973, green: 0.98, blue: 0.988, alpha: 1).cgColor
  cell.layer?.cornerRadius = 4
  let padded = String(format: "%02d", number)
  let value = number % 3 == 0 ? "ok" : number % 3 == 1 ? "warn" : "trace"
  cell.addSubview(label("Runtime check \(padded)", frame: NSRect(x: 4, y: 2, width: 108, height: 12), size: 9, color: muted, bold: false))
  cell.addSubview(label(value, frame: NSRect(x: 124, y: 2, width: 34, height: 12), size: 9, color: NSColor(calibratedRed: 0.059, green: 0.463, blue: 0.431, alpha: 1), bold: true))
  return cell
}

@MainActor
private func orderRow(_ left: String, _ right: String, y: CGFloat, width: CGFloat, bold: Bool = false) -> NSView {
  let row = NSView(frame: NSRect(x: 10, y: y, width: width - 20, height: 15))
  row.addSubview(label(left, frame: NSRect(x: 0, y: 0, width: width - 80, height: 15), size: 11, color: dark, bold: bold))
  row.addSubview(label(right, frame: NSRect(x: width - 82, y: 0, width: 60, height: 15), size: 11, color: dark, bold: true))
  return row
}

@MainActor
private func shell(_ frame: NSRect, padding: CGFloat = 10) -> NSView {
  let view = NSView(frame: frame)
  view.wantsLayer = true
  view.layer?.backgroundColor = NSColor.white.cgColor
  view.layer?.cornerRadius = 10
  view.layer?.borderColor = NSColor(calibratedRed: 0.859, green: 0.89, blue: 0.937, alpha: 1).cgColor
  view.layer?.borderWidth = 1
  return view
}

@MainActor
private func label(_ value: String, frame: NSRect, size: CGFloat, color: NSColor, bold: Bool) -> NSTextField {
  let field = NSTextField(labelWithString: value)
  field.frame = frame
  field.font = bold ? .boldSystemFont(ofSize: size) : .systemFont(ofSize: size)
  field.textColor = color
  field.lineBreakMode = .byTruncatingTail
  field.identifier = NSUserInterfaceItemIdentifier(value.lowercased().replacingOccurrences(of: " ", with: "-"))
  return field
}

private let dark = NSColor(calibratedRed: 0.067, green: 0.094, blue: 0.153, alpha: 1)
private let muted = NSColor(calibratedRed: 0.278, green: 0.333, blue: 0.412, alpha: 1)

File: examples/android-demo/app/src/main/java/ai/uicheck/demo/DemoView.kt
package ai.uicheck.demo

import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.graphics.drawable.GradientDrawable
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView

data class UICheckAndroidDemoView(
  val screen: LinearLayout,
  val title: TextView,
  val summary: LinearLayout,
  val items: LinearLayout,
  val status: LinearLayout,
  val details: LinearLayout,
  val submit: Button
)

fun createUICheckAndroidDemoView(context: Context): UICheckAndroidDemoView {
  val screen = LinearLayout(context).apply {
    id = View.generateViewId()
    orientation = LinearLayout.VERTICAL
    setBackgroundColor(Color.rgb(248, 250, 252))
    contentDescription = "Checkout screen"
  }
  val header = LinearLayout(context).apply {
    orientation = LinearLayout.HORIZONTAL
    setPadding(dp(context, 24), dp(context, 12), dp(context, 24), dp(context, 12))
    setBackgroundColor(Color.rgb(17, 24, 39))
  }
  val headerText = LinearLayout(context).apply {
    orientation = LinearLayout.VERTICAL
  }
  val eyebrow = label(context, "UICheck Android", 13f, Color.rgb(147, 197, 253), true)
  val title = label(context, "Checkout screen", 22f, Color.WHITE, true)
  val badge = TextView(context).apply {
    text = "android"
    textSize = 12f
    setTypeface(typeface, Typeface.BOLD)
    setTextColor(Color.WHITE)
    gravity = android.view.Gravity.CENTER
    setPadding(dp(context, 12), 0, dp(context, 12), 0)
    background = roundedBorder(Color.TRANSPARENT, Color.argb(92, 255, 255, 255), dp(context, 999), dp(context, 1))
  }
  headerText.addView(eyebrow, LinearLayout.LayoutParams(-1, -2))
  headerText.addView(title, LinearLayout.LayoutParams(-1, -2))
  header.addView(headerText, LinearLayout.LayoutParams(0, -2, 1f))
  header.addView(badge, LinearLayout.LayoutParams(-2, dp(context, 32)))

  val content = LinearLayout(context).apply {
    orientation = LinearLayout.VERTICAL
    setPadding(dp(context, 14), dp(context, 14), dp(context, 14), dp(context, 14))
  }
  val summary = card(context, "Registered ref summary", "MCP reads runtime boxes, text, testID and labels.")
  val items = orderCard(context)
  val status = card(context, "Ready for MCP inspection", "This real demo has 100+ inspectable nodes.")
  val details = detailsPanel(context)
  val hint = TextView(context).apply {
    text = "MCP can inspect all elements or a selected target."
    textSize = 11f
    setTypeface(typeface, Typeface.BOLD)
    setTextColor(Color.rgb(15, 118, 110))
    setPadding(dp(context, 10), dp(context, 8), dp(context, 10), dp(context, 8))
    background = roundedFill(Color.rgb(236, 254, 255), dp(context, 10))
  }
  val submit = Button(context).apply {
    id = View.generateViewId()
    text = "Submit order"
    isAllCaps = false
    contentDescription = "Submit order"
    background = roundedFill(Color.rgb(168, 85, 247), dp(context, 10))
    setTextColor(Color.WHITE)
    textSize = 14f
  }

  content.addView(summary, LinearLayout.LayoutParams(-1, dp(context, 66)))
  content.addView(items, LinearLayout.LayoutParams(-1, dp(context, 92)).withTop(context, 8))
  content.addView(status, LinearLayout.LayoutParams(-1, dp(context, 66)).withTop(context, 8))
  content.addView(details, LinearLayout.LayoutParams(-1, dp(context, 370)).withTop(context, 8))
  content.addView(hint, LinearLayout.LayoutParams(-1, dp(context, 34)).withTop(context, 8))
  content.addView(View(context), LinearLayout.LayoutParams(-1, 0, 1f))
  content.addView(submit, LinearLayout.LayoutParams(-1, dp(context, 40)).withTop(context, 8))

  screen.addView(header, LinearLayout.LayoutParams(-1, dp(context, 78)))
  screen.addView(content, LinearLayout.LayoutParams(-1, 0, 1f))
  return UICheckAndroidDemoView(screen, title, summary, items, status, details, submit)
}

private fun card(context: Context, title: String, text: String): LinearLayout =
  shell(context).apply {
    addView(label(context, title, 13f, Color.rgb(17, 24, 39), true), LinearLayout.LayoutParams(-1, dp(context, 18)))
    addView(label(context, text, 11f, Color.rgb(71, 85, 105), false), LinearLayout.LayoutParams(-1, dp(context, 30)))
  }

private fun orderCard(context: Context): LinearLayout =
  shell(context).apply {
    addView(label(context, "Order items", 13f, Color.rgb(17, 24, 39), true), LinearLayout.LayoutParams(-1, dp(context, 18)))
    addView(orderRow(context, "Starter license", "$19"), LinearLayout.LayoutParams(-1, dp(context, 15)))
    addView(orderRow(context, "Team add-on", "$8"), LinearLayout.LayoutParams(-1, dp(context, 15)))
    addView(orderRow(context, "Total", "$27", true), LinearLayout.LayoutParams(-1, dp(context, 16)).withTop(context, 4))
  }

private fun detailsPanel(context: Context): LinearLayout =
  shell(context, padding = 8).apply {
    addView(label(context, "Runtime detail matrix", 13f, Color.rgb(17, 24, 39), true), LinearLayout.LayoutParams(-1, dp(context, 18)))
    val grid = LinearLayout(context).apply { orientation = LinearLayout.VERTICAL }
    for (line in 0 until 17) {
      val row = LinearLayout(context).apply { orientation = LinearLayout.HORIZONTAL }
      for (column in 0 until 2) {
        val number = line * 2 + column + 1
        row.addView(detailCell(context, number), LinearLayout.LayoutParams(0, dp(context, 16), 1f).withRight(context, if (column == 0) 6 else 0))
      }
      grid.addView(row, LinearLayout.LayoutParams(-1, dp(context, 19)))
    }
    addView(grid, LinearLayout.LayoutParams(-1, dp(context, 330)).withTop(context, 4))
  }

private fun detailCell(context: Context, number: Int): LinearLayout {
  val padded = number.toString().padStart(2, '0')
  val value = when (number % 3) {
    0 -> "ok"
    1 -> "warn"
    else -> "trace"
  }
  return LinearLayout(context).apply {
    orientation = LinearLayout.HORIZONTAL
    setPadding(dp(context, 4), dp(context, 1), dp(context, 4), dp(context, 1))
    background = roundedBorder(Color.rgb(248, 250, 252), Color.rgb(226, 232, 240), dp(context, 4), dp(context, 1))
    addView(label(context, "Runtime check $padded", 9f, Color.rgb(51, 65, 85), false), LinearLayout.LayoutParams(0, -1, 1f))
    addView(label(context, value, 9f, Color.rgb(15, 118, 110), true), LinearLayout.LayoutParams(dp(context, 42), -1))
  }
}

private fun orderRow(context: Context, label: String, value: String, strong: Boolean = false): LinearLayout =
  LinearLayout(context).apply {
    orientation = LinearLayout.HORIZONTAL
    addView(label(context, label, 11f, Color.rgb(17, 24, 39), strong), LinearLayout.LayoutParams(0, -1, 1f))
    addView(label(context, value, 11f, Color.rgb(17, 24, 39), true), LinearLayout.LayoutParams(dp(context, 60), -1))
  }

private fun shell(context: Context, padding: Int = 10): LinearLayout =
  LinearLayout(context).apply {
    id = View.generateViewId()
    orientation = LinearLayout.VERTICAL
    val resolvedPadding = dp(context, padding)
    setPadding(resolvedPadding, resolvedPadding, resolvedPadding, resolvedPadding)
    background = roundedBorder(Color.WHITE, Color.rgb(219, 227, 239), dp(context, 10), dp(context, 1))
  }

private fun roundedFill(color: Int, radius: Int): GradientDrawable =
  GradientDrawable().apply {
    setColor(color)
    cornerRadius = radius.toFloat()
  }

private fun roundedBorder(color: Int, strokeColor: Int, radius: Int, strokeWidth: Int): GradientDrawable =
  GradientDrawable().apply {
    setColor(color)
    cornerRadius = radius.toFloat()
    setStroke(strokeWidth, strokeColor)
  }

private fun label(context: Context, value: String, size: Float, color: Int, bold: Boolean): TextView =
  TextView(context).apply {
    text = value
    textSize = size
    setTextColor(color)
    if (bold) setTypeface(typeface, Typeface.BOLD)
    includeFontPadding = false
  }

private fun dp(context: Context, value: Int): Int =
  (value * context.resources.displayMetrics.density).toInt()

private fun LinearLayout.LayoutParams.withTop(context: Context, value: Int): LinearLayout.LayoutParams {
  topMargin = dp(context, value)
  return this
}

private fun LinearLayout.LayoutParams.withRight(context: Context, value: Int): LinearLayout.LayoutParams {
  rightMargin = dp(context, value)
  return this
}
